### ISSUE TITLE
Inject a generic supertype interface.

### ISSUE THREAD

**Issue by kevind-wgu** [avaje/avaje-http#84]
After playing with dagger and being frustrated that I was not able to inject interfaces without manually binding each one I was pleasantly surprised to find that I could inject interfaces in the Avaje framework. This only works for the first interface though and not interfaces that the first interface inherit. Some example code.

```
public interface ReadService<T, KeyType> {
    Optional<T> get(KeyType id);
}
...

public interface CRUDService<T, KeyType> extends
        ReadService<T, KeyType>,
        CreateService<T, KeyType>,
        UpdateService<T>,
        DeleteService<KeyType>
{
}

public class MyCRUDService implements CRUDService<MyObj, Integer> {
 ...
}
```

Given the above code I can inject `MyCrudService` and `CRUDService<MyObj, Integer>` but I am unable to inject `ReadService<MyObj, Integer>`. I don't know the work it would take to be able to do that. I know that Micronaut allows me to do it and thought it would be useful here as well.

**Comment by rbygrave** [comment]
Yeah cool.  We ought to be able to inject any of the 5 interfaces that MyCRUDService implements.  I suspect this is related to the generic type handling which I'm now looking to simplify with that other bug.

I'll look to add a failing test for this and see what's up with this case.

**Comment by rbygrave** [comment]
Logged in avaje-inject as https://github.com/avaje/avaje-inject/issues/228

**Comment by rbygrave** [comment]
Fixed in avaje-inject version 8.6

**Comment by rbygrave** [comment]
avaje-inject version 8.6 has been released.  I'll close this and reopen if necessary.
